### PR TITLE
opencv, visp, vtk@8.2: fix up Cellar path references

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

See Homebrew/discussions#1819. Marking as `do not merge` first since I'll want to take a peek at the bottles to check how much this resolves things.

### CODE REVIEW

**Comment by carlocab** [comment]
This got rid of a lot of the Cellar references, but a few remain. Will probably need to patch some generated files.

File: Formula/visp.rb [hunk 1]
```diff
@@ -83,7 +83,20 @@ def install
...
+      "unix-install/vpConfig.h",
+      "unix-install/VISPConfig.cmake",
+      "VISPConfig.cmake",
+      "CMakeCache.txt",
+    ]
+    inreplace opencv_references, opencv.prefix.realpath, opencv.opt_prefix
     system "make", "install"
+
+    # Make sure software built against visp don't reference opencv's cellar path either
+    inreplace lib/"pkgconfig/visp.pc", opencv.prefix.realpath, opencv.opt_prefix
```
Comment (carlocab) [inline Formula/visp.rb:99]: This might error out now that I've done a bunch of `inreplace`s above, but let's see.

**Comment by carlocab** [comment]
Unfortunately, there is a single remaining reference to `opencv`'s cellar left in `visp`:
```
❯ strings -t x - libvisp_vision.3.4.0.dylib | rg 'Cellar/opencv'
8131a /usr/local/Cellar/opencv/4.5.3_1/include/opencv4/opencv2/core/mat.inl.hpp
```
This is a substantial improvement from before. An alternative to this is that we just embed `opencv`'s `pkg_version` in `visp`'s keg during the build and then test that this matches the `opencv` in use in the `test` block.

**Comment by SMillerDev** [comment]
We could do 
```
strings -t x - libvisp_vision.#{version}.dylib | rg "#{Formula["OpenCV"].prefix}"
```

If it fails we have a problem.

As for the inreplace statements, is there no way to specify in a `./configure` type way where it should look for these paths?

**Comment by carlocab** [comment]
> As for the inreplace statements, is there no way to specify in a `./configure` type way where it should look for these paths?

I looked, and it doesn't seem like it, unfortunately. I think that last reference might be safe to ignore though. Or we can still just make sure `visp`'s test fails whenever `opencv`'s version/revision changes so that CI reminds us to give it a revision bump.

We probably still want to minimise `Cellar` references tho to avoid anyone compiling anything with `visp` having their stuff break from `opencv` version/revision bumps.

**Comment by carlocab** [comment]
Included in #81244.

**Pull request closed without merging**